### PR TITLE
Added NodeSelection functionality

### PR DESCRIPTION
**Pull request by bpickard22**
Added NodeSelection functionality to gather_network_logs

Users may now pass in nodes that exist on the cluster to the gather and
it will only run on those nodes.
User input is verified against nodes that exist on the cluster
To use run $oc adm must-gather -- gather_network_logs <node/s> 
Signed-off-by: Ben Pickard <bpickard@redhat.com>

### CODE REVIEW

**Comment by jluhrsen** [comment]
nothing stood out to me looking at the code.
Is there any example using this in some doc or maybe we don't need that?

**Comment by bpickard22** [comment]
> nothing stood out to me looking at the code.
> Is there any example using this in some doc or maybe we don't need that?
@jluhrsen 
added use command to comment of pr. Is that a good place for it or would you want a document with some examples?

**Comment by jluhrsen** [comment]
> > nothing stood out to me looking at the code.
> > Is there any example using this in some doc or maybe we don't need that?
> > @jluhrsen
> > added use command to comment of pr. Is that a good place for it or would you want a document with some examples?

yeah sure. I was expecting some other place with some docs about how to use different gather commands, but I'm not finding anything where you'd add this to.

File: collection-scripts/gather_network_logs [hunk 1]
```diff
@@ -228,21 +228,41 @@ function gather_kuryr_data {
...
+TMP_ARR=() #tmp array to place the args that are nodes when they are checked against the cluster nodes
+if [ $# -eq 0 ]; then 
     echo "WARNING: Collecting network logs on ALL linux nodes in your cluster. This could take a long time." >&2
+else
+    for (( i=0; i<ELEMENTS; i++));do
+      for (( j=0; j<NODE_LENGTH; j++));do
+        if [[ "${ARGS[$i]}" == "${NODES[$j]}" ]]; then
+        echo  ${ARGS[$i]} "->node exists in nodes"
+        TMP_ARR[${#TMP_ARR[@]}]+="${NODES[$j]}"
+        fi
```
Comment (flavio-fernandes) [inline collection-scripts/gather_network_logs:253]: NIT: should we have a warning for node names that were provided by caller and are not found from "oc get nodes"?

Comment (bpickard22) [reply collection-scripts/gather_network_logs:253]: probably would be a good thing to add

Comment (astoycos) [reply collection-scripts/gather_network_logs:253]: +1 here

File: collection-scripts/gather_network_logs [hunk 2]
```diff
@@ -228,21 +228,41 @@ function gather_kuryr_data {
...
+NODE_LENGTH=${#NODES[@]}
+TMP_ARR=() #tmp array to place the args that are nodes when they are checked against the cluster nodes
+if [ $# -eq 0 ]; then 
     echo "WARNING: Collecting network logs on ALL linux nodes in your cluster. This could take a long time." >&2
+else
+    for (( i=0; i<ELEMENTS; i++));do
+      for (( j=0; j<NODE_LENGTH; j++));do
+        if [[ "${ARGS[$i]}" == "${NODES[$j]}" ]]; then
+        echo  ${ARGS[$i]} "->node exists in nodes"
+        TMP_ARR[${#TMP_ARR[@]}]+="${NODES[$j]}"
```
Comment (astoycos) [inline collection-scripts/gather_network_logs:252]: Why do you need the TMP_ARR array here? Just remove it from the `ARGS` array if it doesn't exist in `NODES
`

**Comment by sferich888** [comment]
@bpickard22 have you had a chance to checkup on this?

File: collection-scripts/gather_network_logs [hunk 3]
```diff
@@ -228,21 +228,41 @@ function gather_kuryr_data {
 
 function gather_kuryr_nodes_data {
   echo "INFO: Gathering kuryr nodes data"
-  for NODE in ${NODES}; do
-      CNI_POD=$(oc -n openshift-kuryr get pods --no-headers -o custom-columns=":metadata.name" --field-selector spec.nodeName="${NODE}" -l app=kuryr-cni)
+  for NODE in "${NODES[@]}"; do
+      CNI_POD=($(oc -n openshift-kuryr get pods --no-headers -o custom-columns=":metadata.name" --field-selector spec.nodeName="${NODE}" -l app=kuryr-cni))
```
Comment (RickJWagner) [inline collection-scripts/gather_network_logs:232]: What difference is the extra paragraph making?  
CNI_POD=($(oc -n           << instead of  CNI_POD=$(oc -n
not criticizing, just wondering why this change is necessary.

File: collection-scripts/gather_network_logs [hunk 4]
```diff
@@ -228,21 +228,41 @@ function gather_kuryr_data {
...
           nsenter --net=/host_proc/\$pid/ns/net ip -d addr; \
           done" > "${NETWORK_LOG_PATH}"/"${CNI_POD}"_interfaces & PIDS+=($!)
   done
 }
-
-if [ $# -eq 0 ]; then
+ 
+ARGS=($@) #args passed in by the user
+IFS=" " read -r -a NODES <<< $(oc get nodes -o jsonpath='{range .items[*]}{@.metadata.name} {.status.nodeInfo.operatingSystem==linux}')
+ELEMENTS=${#ARGS[@]} #length of args passed in by the user
```
Comment (RickJWagner) [inline collection-scripts/gather_network_logs:242]: Thanks for leaving the explanatory comments, this makes reviews easier and will be helpful to future readers, too.

**Review by RickJWagner — COMMENTED** [review]
Left a few comments in-line with code.

**Comment by bpickard22** [comment]
closing this pr due to the deprecation of gather_network_logs. If we decide that we want to revisit this idea in must_gather after the refactor I can take care of it there
